src/reconcile/rec_write.c

```diff
@@ -1407,6 +1406,20 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
 		} else
 			upd_select->start_ts =
 			    upd_select->stop_ts = upd_select->upd->start_ts;
+
+		/*
+		 * Finalize the timestamps, checking if the update is globally
+		 * visible and timestamps won't need to be written.
+		 *
+		 * TIMESTAMP-FIXME
+		 * Values (presumably) have associated transaction IDs, but we
+		 * haven't yet decided how to handle them. For now, ignore them
+		 * in determining value durability.
+		 */
+		if (upd_select->start_ts != WT_TS_NONE &&
+		    __wt_txn_visible_all(
+		    session, WT_TXN_NONE, upd_select->start_ts))
+			upd_select->start_ts = WT_TS_NONE;
```

Comment:
I'm uneasy about reconciliation (or any code) changing anything in a committed update structure.  We have a lot of assumptions in many places of the code that committed updates are immutable.

That said, there is already a FIXME here, and I expect the real solution will be to create a `WT_UPDATE_VALIDITY` struct (or similar) that will hold the relevant information (start and stop timestamps, the latter should be calculated here and removed from `WT_UPDATE`, along with transaction IDs).  That should mean reconciliation is not making any modifications to updates.